### PR TITLE
ConvertTo-Json: Serialize BigInteger as a number

### LINKED ISSUE #20989: Serializing bigint value to JSON emits incorrect JSON
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

```ps1
PS> $i = [bigint]"93839802329326"
PS> $obj = @{"val" = $i}
PS> ConvertTo-Json $obj 

{
    "val":  {
                "IsPowerOfTwo":  false,
                "IsZero":  false,      
                "IsOne":  false,       
                "IsEven":  true,
                "Sign":  1
            }
}
```

Note that casting the variable inline as `[int64]` does work correctly:

```ps1
PS> $i = [bigint]"93839802329326"
PS> $obj = @{"val" = [int64]$i}
PS> ConvertTo-Json $obj 
{
    "val":  93839802329326
}
```

and casting the variable inline as `[bigint]` does not:

```ps1
PS> $i = [bigint]"93839802329326"
PS> $obj = @{"val" = [bigint]$i}
PS> ConvertTo-Json $obj 
{
  "val": {
    "IsPowerOfTwo": false,
    "IsZero": false,
    "IsOne": false,
    "IsEven": true,
    "Sign": 1
  }
}
```

### Expected behavior

```console
{
    "val":  93839802329326
}
```

### Actual behavior

```console
{
    "val":  {
                "IsPowerOfTwo":  false,
                "IsZero":  false,      
                "IsOne":  false,       
                "IsEven":  true,
                "Sign":  1
            }
}
```

### Error details

_No response_

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.4.0
PSEdition                      Core
GitCommitId                    7.4.0
OS                             Microsoft Windows 10.0.19045
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by jborean93**
# PR Summary
Serializes a BigInteger value as a raw JSON number rather than the object's properties which does not contain the BigInteger value.

This is technically a breaking change, I went ahead with the PR anyway to see if it was actually possible to do (it is). Waiting for the triage outcome of https://github.com/PowerShell/PowerShell/issues/20989#issuecomment-1876298384 to see if this breaking change is ok or not.

## PR Context
Fixes: https://github.com/PowerShell/PowerShell/issues/20989

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs [hunk 1]
```diff
@@ -519,6 +520,10 @@ private static object ProcessValue(object obj, int currentDepth, in ConvertToJso
             {
                 rv = jObject.ToObject<Dictionary<object, object>>();
             }
+            else if (obj is BigInteger bi)
+            {
+                rv = new JRaw(bi.ToString(NumberFormatInfo.InvariantInfo));
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:525]: Why do we need `JRaw`?

Comment (jborean93) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:525]: So the value isn’t quoted and is a literal numeric value.

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:525]: Perhaps we should use `string` if we agree with https://github.com/PowerShell/PowerShell/pull/20999#issuecomment-1878187217

Comment (jborean93) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs:525]: Personally I don't see why we should but I'll leave it up to the WG decision in the end.

Comment (jborean93) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs]: Turns out it isn't needed we just need to ensure the value is the BigInteger itself and we don't do PowerShell's property enumerator.

**Review by iSazonov — APPROVED** [review]

**Review by SeeminglyScience — APPROVED** [review]
LGTM! Thanks @jborean93!

**Pull request merged**